packages/structures/src/Structure.ts

```diff
@@ -0,0 +1,139 @@
+	 *
+	 * @internal
+	 */
+	protected [kData]: Readonly<ReplaceOmittedWithUnknown<Omitted, DataType>>;
+
+	/**
+	 * Creates a new structure to represent API data
+	 *
+	 * @param data - the data from the API that this structure will represent
+	 * @remarks To be made public in subclasses
+	 * @internal
+	 */
+	public constructor(data: Readonly<Partial<DataType>>, ..._rest: unknown[]) {
+		this[kData] = Object.assign(this.getDataTemplate(), data);
+		this[kMixinConstruct]?.(data);
+	}
+
+	/**
+	 * Patches the raw data of this object in place
+	 *
+	 * @param data - the updated data from the API to patch with
+	 * @remarks To be made public in subclasses
+	 * @returns this
+	 * @internal
+	 */
+	protected [kPatch](data: Readonly<Partial<DataType>>): this {
+		this[kData] = Object.assign(this.getDataTemplate(), this[kData], data);
+		this.optimizeData(data);
+		return this;
+	}
+
+	/**
+	 * Creates a clone of this structure
+	 *
+	 * @returns a clone of this
+	 * @internal
+	 */
+	protected [kClone](patchPayload?: Readonly<Partial<DataType>>): typeof this {
+		const clone = this.toJSON();
+		// @ts-expect-error constructor is of abstract class is unknown
+		return new this.constructor(patchPayload ? Object.assign(clone, patchPayload) : clone);
+	}
+
+	/**
+	 * Function called to ensure stored raw data is in optimized formats, used in tandem with a data template
+	 *
+	 * @example created_timestamp is an ISO string, this can be stored in optimized form as a number
+	 * @param _data - the raw data received from the API to optimize
+	 * @remarks Implementation to be done in subclasses and mixins where needed.
+	 * For typescript users, mixins must use the closest ancestors access modifier.
+	 * @remarks Automatically called in Structure[kPatch] but must be called manually in the constructor
+	 * of any class implementing this method.
+	 * @remarks Additionally, when implementing, ensure to call `super._optimizeData` if any class in the super chain aside
+	 * from Structure contains an implementation.
+	 * Note: mixins do not need to call super ever as the process of mixing walks the prototype chain.
+	 * @virtual
+	 * @internal
+	 */
+	protected optimizeData(_data: Partial<DataType>) {}
+
+	/**
+	 * Transforms this object to its JSON format with raw API data (or close to it),
+	 * automatically called by `JSON.stringify()` when this structure is stringified
+	 *
+	 * @remarks
+	 * The type of this data is determined by omissions at runtime and is only guaranteed for default omissions
+	 * @privateRemarks
+	 * When omitting properties at the library level, this must be overridden to re-add those properties
+	 */
+	public toJSON(): DataType {
+		// This will be DataType provided nothing is omitted, when omits occur, subclass needs to overwrite this.
+		const data = (
+			Object.values(this[kData]).some((value) => typeof value === 'object' && value !== null)
+				? structuredClone(this[kData])
+				: { ...this[kData] }
```

Comment:
What exactly is the point of all this logic? Why would we use structuredClone just if we have nested objects? Not a blocker for the PR but I wanna know XD